tests/sparseml/transformers/obcq/test_obcq_completion.py

```diff
@@ -0,0 +1,158 @@
+GPU_CONFIGS_DIRECTORY = "tests/sparseml/transformers/obcq/obcq_configs/completion/gpu"
+
+
+@requires_torch
+@pytest.mark.integration
+@parameterized_class(parse_params(CONFIGS_DIRECTORY))
+class TestOBCQCompletion(unittest.TestCase):
+    """
+    Test for oneshot for quantization and quantization + sparsity. Sparsity-only tests
+    can be found under `test_obcq_sparsity.py`
+    """
+
+    model = None
+    dataset = None
+    recipe = None
+    sparsity = None
+
+    def setUp(self):
+        import torch
+
+        self.device = "cuda:0" if torch.cuda.is_available() else "cpu"
+        self.output = "./oneshot_output"
+
+    def labeled_dataloader(self, dataset_name, model_name):
+        from torch.utils.data import DataLoader
+        from transformers import DefaultDataCollator
+
+        from sparseml.transformers import SparseAutoTokenizer
+        from sparseml.transformers.finetune.data import TextGenerationDataset
+        from sparseml.transformers.finetune.data.data_args import DataTrainingArguments
+
+        tokenizer = SparseAutoTokenizer.from_pretrained(model_name)
+        data_args = DataTrainingArguments(
+            dataset=dataset_name,
+            max_seq_length=512,
+            pad_to_max_length=False,
+        )
+        dataset_manager = TextGenerationDataset.load_from_registry(
+            data_args.dataset,
+            data_args=data_args,
+            split="train",
+            tokenizer=tokenizer,
+        )
+        calib_dataset = dataset_manager.tokenize_and_process(
+            dataset_manager.get_raw_dataset()
+        )
+        data_loader = DataLoader(
+            calib_dataset, batch_size=1, collate_fn=DefaultDataCollator()
+        )
+
+        return data_loader
+
+    def test_oneshot_completion(self):
+        import torch
+
+        from sparseml.pytorch.model_load.helpers import get_session_model
+        from sparseml.pytorch.utils import tensors_to_device
+        from sparseml.transformers import oneshot
+
+        oneshot(
+            model=self.model,
+            dataset=self.dataset,
+            oneshot_device=self.device,
+            recipe=self.recipe,
+            max_seq_length=128,
+            num_calibration_samples=32,
+            pad_to_max_length=False,
+            output_dir=self.output,
+            clear_sparse_session=False,
+        )
+
+        first_tiny_model = get_session_model()
+
+        dataset = "open_platypus"
+
+        iter = 10
+        dataloader = self.labeled_dataloader(dataset, self.model)
+        total_new_ppl = 0.0
+        for idx, sample in enumerate(dataloader):
+            if idx >= iter:
+                break
+            new_output = first_tiny_model(**(tensors_to_device(sample, self.device)))
+            new_ppl = torch.exp(new_output.loss)
+            total_new_ppl += new_ppl
+
+        avg_new_ppl = total_new_ppl / iter
+        self.assertLess(avg_new_ppl, 5000)
```

Comment:
Could we make the expected perplexity an entry in the config? I don't think it makes sense to hardcode since it differs depending on the model

Reply:
Does this run for the 7b model or is that in a different test? We may want to run with more calibration samples and more test samples for the 7b model so we can better track if the ppl degrades. 5,000 is way too high a threshold for 7b and should be below 20 at least